torch/csrc/autograd/python_hook.cpp

```diff
@@ -0,0 +1,42 @@
+#include "torch/csrc/autograd/python_hook.h"
+
+#include "THP.h"
+#include "torch/csrc/autograd/python_variable.h"
+#include "torch/csrc/utils/auto_gil.h"
+#include "torch/csrc/utils/object_ptr.h"
+#include "torch/csrc/Exceptions.h"
+
+namespace torch { namespace autograd {
+
+PyGradHook::PyGradHook(PyObject* dict) : dict(dict) {
+  Py_INCREF(dict);
```

Comment:
Is there any reason why you get the GIL for every operation on `PyGradHook` except for creation?